Assorted controls fixes for macOS 26 Tahoe

vslavik opened the pull request:
This PR covers the rest of Tahoe breakages I am aware of (via Poedit). It is best reviewed individually, see the commits' explanations.

This most prominently fixes single-line `wxTextCtrl`, which was outright invisible on macOS 26 Contrast-Is-For-Amateurs Tahoe. `wxSpinCtrl` was subtly broken too, but that's a much less prominent control.

The big commit is the 1st one, [Don't set NSView.clipsToBounds = YES globally](https://github.com/wxWidgets/wxWidgets/commit/3273d289d7ca710a1fd4833c748facbc248e367b) — this slightly undoes the changes to fix wxAUI rendering (and no doubt much more, #23916 was just the canary), by limiting  the `clipsToBounds` change to non-native windows only.

The alternative would be to do it individually for controls that are known to be mis-rendered, but I somewhat strongly prefer doing it like this: we should mess with native controls as little as possible. Unless I'm missing something (@csomor?) it should be safe to differentiate native-vs-wx with this in `wxWidgetCocoaImpl::wxWidgetCocoaImpl`:
https://github.com/wxWidgets/wxWidgets/blob/3273d289d7ca710a1fd4833c748facbc248e367b/src/osx/cocoa/window.mm#L2738-L2739

This assumption is crucial; if I'm wrong about that, it should be done per-control (i.e. so far at least for `wxTextCtrl` and `wxSpinButton`).

Screenshot before:

<img width="644" height="594" alt="before_full" src="https://github.com/user-attachments/assets/47835469-96a1-47e4-895e-bcb06341e26d" />

And after:

<img width="644" height="588" alt="after_full" src="https://github.com/user-attachments/assets/85863d49-43c8-4b68-b4bb-a49512628851" />

And  just the spin control without focus ring:

Before:

<img width="154" height="48" alt="before_spin" src="https://github.com/user-attachments/assets/2f9249ab-c0ff-4564-9fb1-fdb81335d95b" />

After:

<img width="158" height="43" alt="after_spin" src="https://github.com/user-attachments/assets/9f0bfa3c-30fa-4eb5-9d45-c82a8d5b2508" />

In the process of doing this, I spotted some inaccuracies on macOS 15 too, mostly in sizing, so I fixed them there too; it is obvious from the commits where.

---

For [Fix wxTextCtrl sizes on macOS, particularly 26](https://github.com/wxWidgets/wxWidgets/commit/26ee6d248b6e5c96f811ffb34a67c73168157643) my methodology for verifying sizes was as follows:
1. Create test view in wx for measuring, and also create a native xib in Interface Builder, using Xcode 16 for macOS 15 and Xcode 26 for Tahoe, to avoid Xcode 26's metrics changes.
2. I then measures heights of focused text controls with xScope, and also verified regular and small changes against controls present in Apple Safari and Apple Mail.
3. After verifying all sizes were correct, I commented out the code to call `GetPeer()->GetBestSize()` to make sure the fallback switch statement was executed, and again measured all controls on both macOS versions.

Which is to say that I'm reasonably confident in them.

vslavik committed:
Fix wxCollapsiblePane button size on macOS

Increase the size so that it is full rendered including its shadows.
Previously, shadows were subtly cut off on macOS 11+ as well as
macOS 26.

vslavik committed:
Don't set NSView.clipsToBounds = YES globally

In 991a74c wxWidgetImpl was changed to set clipsToBounds = YES globally,
to prevent new behavior in macOS 14, where views' rendering can extend
outside the view area, from applying to wxWidgets. This was to prevent
rendering artifacts (see e.g. #23916) because wxWindow rendering being
clipped to its area is arguably part of wxWindow API contract.

This however breaks rendering of several controls (most prominently
single-line wxTextCtrl) on macOS 26 Tahoe, and possibly more subtly
earlier too.

To avoid this, change the code to force clipsToBounds only on non-native
wxWindows and leave it as-is (which is false on macOS >= 14) for native
ones, so that their rendering isn't disrupted.

Also add a semi-private wxWindow::MacClipsToBounds() method to allow
customizing this behavior. This is important for e.g. generic composite
controls that may be packed too tight and clip their children.

This fixes broken rendering of wxTextCtrl and wxSpinButton on macOS 26.

vslavik committed:
Improve wxSpinCtrl sizing and appearance on macOS

Multiple improvements to wxSpinCtrl/wxSpinButton appearance:

- fixes broken rendering on macOS 26 (clipped borders)

- corrects wxSpinButton sizes (note that NSStepper doesn't dynamically
  adjust to the size it is set to, but renders at a constant size)

- adds typically used margin between the stepper and the text field

- removes unreachable code in wxWindowMac::DoGetBestSize

vslavik committed:
Improve default button size on macOS

wxOSX's default was a bit too larger on at least macOS 11+.

The new value is still slightly larger than what Interface Builder
creates by default, but matches the size used in native file open
windows (on macOS 26, it is 1pt larger on macOS 15, but that's not
noticeable and not worth complicating the code for).

Note that native AppKit apps often use even smaller button sizes.

vslavik committed:
Fix wxTextCtrl sizes on macOS, particularly 26

Slightly tweak hardcoded sizes to correspond to natively used sizes.
Note that this code is only used in e.g. generic controls whrere native
peer view isn't available yet; normally sizing is done natively.

Adjust the sizes for macOS 26, where text fields are larger.

Make sure the size is never smaller than NSMiniControlSize one; for some
reason wxNSTextFieldControl::GetBestSize() did calculate slightly
smaller fitting size. The mini size is so small that using it as hard
cap seems reasonable.

csomor reviewed (approved):
Thanks @vslavik, yes, I think it's fine to go ahead, and collect feedback whether we might have to fine tune because there is some nesting of controls that might bite us

csomor commented:
one comment: when running an app that was built on earlier SDK and is now running on Tahoe, I think, the view is different, ie 'old', so I'd prefer to bracket the `WX_IS_MACOS_AVAILABLE` with the `__MAC_OS_X_VERSION_MAX_ALLOWED `

```
#if __MAC_OS_X_VERSION_MAX_ALLOWED >= 260000
    if ( WX_IS_MACOS_AVAILABLE(26, 0) ) 
```

vadz commented on src/osx/cocoa/window.mm:
```diff
@@ -2734,7 +2734,9 @@ void wxOSXCocoaClassAddWXMethods(Class c, wxOSXSkipOverrides skipFlags)
     [m_osxView release];
-    m_osxView.clipsToBounds = YES;
+
+    if ( flags & Widget_IsUserPane )
```
Very minor, but if you amend this to use `IsUserPane()` instead, please also do `s/they're/their/` in the commit message as this triggers my OCD :-/

vslavik commented:
> one comment: when running an app that was built on earlier SDK and is now running on Tahoe, I think, the view is different, ie 'old', so I'd prefer to bracket the WX_IS_MACOS_AVAILABLE with the __MAC_OS_X_VERSION_MAX_ALLOWED 

You're right, thanks! I assumed this wouldn't trigger if compiled with older compiler, but it does. That's a problem, because code compiled with Xcode pre-26 uses old metrics and appearance.

csomor reviewed (approved):
apart from the build-time condition I mentioned, this looks good to me!
@vslavik, thanks a lot for investing your time on these important details

vslavik committed:
Guard macOS 26 checks with __MAC_OS_X_VERSION_MAX_ALLOWED

Wrap uses of WX_IS_MACOS_AVAILABLE(26, 0) with compile-time
__MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_VERSION_26_0 checks so that
code compiled with older versions of Xcode 26 is not affects.

macOS 26 doesn't apply its new appearance to applications that weren't
compiled with Xcode 26+, so this is the correct thing to do: we do not
want any adjustments to apply to 26-unaware code.

vadz commented:
Merged, thanks again!
